src/test/java/org/plp/implementations/plpisa/PlpFileTest.java

```diff
@@ -32,13 +38,106 @@ void createPlpFileSuccessFulTest() {
         Path testOSXFile = testOSXFileSystem.getPath("test.asm");
         Assertions.assertDoesNotThrow(() -> new PlpFile(testOSXFile));
 
-        Path testUnixFile = testOSXFileSystem.getPath("test.asm");
+        Path testUnixFile = testUnixFileSystem.getPath("test.asm");
         Assertions.assertDoesNotThrow(() -> new PlpFile(testUnixFile));
 
-        Path testWinFile = testOSXFileSystem.getPath("test.asm");
+        Path testWinFile = testWinFileSystem.getPath("test.asm");
         Assertions.assertDoesNotThrow(() -> new PlpFile(testWinFile));
     }
 
+    @Test
+    void createPlpFileSuccessFulForExistingFileTest() throws IOException{
+        List<String> programLines = new ArrayList<>();
+        programLines.add(".org 0x100000");
+        programLines.add("");
+        programLines.add("add $r1,$r2  # this is a test");
+
+        Path testOSXFile = testOSXFileSystem.getPath("test.asm");
+        Files.createFile(testOSXFile);
+        Files.write(testOSXFile, programLines);
+        PlpFile plpOsxFile = new PlpFile(testOSXFile);
+        Assertions.assertEquals(plpOsxFile.getInstructions(), programLines);
+
+
+        Path testUnixFile = testUnixFileSystem.getPath("test.asm");
+        Files.createFile(testUnixFile);
+        Files.write(testUnixFile, programLines);
+        PlpFile plpUnixFile = new PlpFile(testUnixFile);
+        Assertions.assertEquals(plpUnixFile.getInstructions(), programLines);
+
+        Path testWinFile = testWinFileSystem.getPath("test.asm");
+        Files.createFile(testWinFile);
+        Files.write(testWinFile, programLines);
+        PlpFile plpWinFile = new PlpFile(testWinFile);
+        Assertions.assertEquals(plpWinFile.getInstructions(), programLines);
+    }
+
+    @Test
+    void getInstructionAtLineNumberFromPlpFileTest() throws IOException{
+        List<String> programLines = new ArrayList<>();
+        programLines.add(".org 0x100000");
+        programLines.add("");
+        programLines.add("add $r1,$r2  # this is a test");
+
+        Path testOSXFile = testOSXFileSystem.getPath("test.asm");
+        Files.createFile(testOSXFile);
+        Files.write(testOSXFile, programLines);
+        PlpFile plpOsxFile = new PlpFile(testOSXFile);
+        Assertions.assertEquals(".org 0x100000", plpOsxFile.getInstructionAtLine(1));
```

Comment:
Make a for loop?

Assertions.assertEquals( programLines.get(index), plpOsxFile.getInstructionAtLine(index + 1);

Reply:
Yeah I could have done that.